src/modules/builtin/len.rs

```diff
@@ -50,21 +50,11 @@ impl SyntaxModule<ParserMetadata> for Len {
 }
 
 impl TranslateModule for Len {
-    fn translate(&self, meta: &mut TranslateMetadata) -> String {
+    fn translate(&self, meta: &mut TranslateMetadata) -> TranslationFragment {
         let value = self.value.translate(meta);
-        if self.value.get_type() == Type::Text {
-            meta.stmt_queue.push_back(format!("__AMBER_LEN={value}"));
-            return String::from("\"${#__AMBER_LEN}\"")
-        }
-        // Case for Array passed as a reference
-        if value.starts_with("\"${!") {
-                meta.stmt_queue.push_back(format!("__AMBER_LEN=({value})"));
-                String::from("\"${#__AMBER_LEN[@]}\"")
-        } else {
-            format!("\"${{#{}", value.trim_start_matches("\"${"))
-                .trim_end()
-                .to_string()
-        }
+        let id = meta.gen_value_id();
+        let var_fragment = meta.push_stmt_variable("__length", Some(id), Type::Num, value).set_get_length();
```

Comment:
So we are not using anymore AMBER_LEN variable?

Reply:
It's being used internally by the `VarFragment` which now handles getting length in case of `Text` and `Array` and also cases for when the variable is a reference. I also decided to shorten the variable names too (we can revert the change if we want to). I just think that this way the output code looks cleaner.